add photils plugin

scheckmedia opened the pull request:
A darktable plugin that tries to predict keywords based on the selected image. This plugin uses photils-cli to handle this task. Photils-cli is an application that passes the image through a neural network, classifies it, and extracts the suggested tags. Everything happens offline without the need that your data are sent over the internet.

[Further details at discuss.pixls.us](https://discuss.pixls.us/t/plugin-for-creating-automatic-tags/18167)

supertobi commented on contrib/photils.lua:
```diff
@@ -0,0 +1,414 @@
+
+function PHOTILS.tagged_image_has_changed()
+    GUI.warning.label = _(
+                            "The suggested tags were not generated\n for the currently selected image!")
```
Why is here a line brake.

supertobi commented on contrib/photils.lua:
```diff
@@ -0,0 +1,414 @@
+        GUI.next_button.sensitive = true
+    end
+
+    local offset = ((PHOTILS.page - 1) * PHOTILS.per_page) + 1
```
A comment why you are doing this offset calculation would not hurt.

supertobi commented on contrib/photils.lua:
```diff
@@ -0,0 +1,414 @@
+    dt.print(_("Tags successfully attached to image"))
+end
+
+function PHOTILS.get_tmp_file()
```
@wpferguson I think is some thing we should add to lib/dtutils.file. What do you think?

wpferguson replied:
@supertobi I agree.  I think I would name it create_tmp_file, since it creates a tmp file.

supertobi replied:
@scheckmedia Then please rename get_tmp_file() to create_tmp_file() and move it to lib/dtutils.file and then use it here, thanks.

supertobi commented on contrib/photils.lua:
```diff
@@ -0,0 +1,414 @@
+    local executable = photils_installed
+
+    if dt.configuration.running_os == "macos" then
+        executable =  executable .. "/Contents/MacOS/photils-cli"
```
I have no clue about macOS. Is it save to use a fixed path?

wpferguson replied:
@supertobi I think this is safe since @scheckmedia controls the executable and where it's installed.

supertobi commented:
Works faster then expected, better results then expected and the photils-cli is smaller then expected. Very nice.

One small blocker, please write all labels in lower case to integrate it better in darktable. 

And I have three ideas of improvements. But nothing that blocks from merging.
1. Add the info, that photils-cli does not work on windows.
2. Perhaps an option to auto update the tags, when selecting a new image.
3. Perhaps show the confidence value in brackets behind the tags to get a feeling for this number.

scheckmedia commented:
I've changed to labels to lower case as suggested, looks better :)

1. I've added a small comment in the source code
2. I've implemented this function but the function slows down darktable a lot, so I will keep it out. A nice idea but unfortunately not possible at the moment.
3. it's implemented now

supertobi commented:
Thanks for the nice script. I've merged it. Feel free to send new pull requests for further improvements.
